src/rust/bitbox02/src/memory.rs

```diff
@@ -12,6 +13,20 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// corresponds to C MEMORY_DEVICE_NAME_MAX_LEN - 1 (no null terminator).
+pub const DEVICE_NAME_MAX_SIZE: usize = 63;
+
+/// `name.as_bytes()` must be smaller or equal to
+/// `DEVICE_NAME_MAX_SIZE`, otherwise this function panics.
+pub fn set_device_name(name: &str) -> Result<(), ()> {
+    match unsafe {
+        bitbox02_sys::memory_set_device_name(crate::str_to_cstr_force!(name, 63).as_ptr())
```

Comment:
@NickeZ I always wondered why I can't use constants in this macro:

```
error: no rules expected the token `DEVICE_NAME_MAX_SIZE`
  --> ../src/rust/bitbox02/src/memory.rs:24:45
   |
24 |             crate::str_to_cstr_force!(name, DEVICE_NAME_MAX_SIZE).as_ptr(),
   |                                             ^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
   | 
  ::: bitbox02/src/util.rs:80:1
   |
80 | macro_rules! str_to_cstr_force {
   | ------------------------------ when calling this macro

error: no rules expected the token `DEVICE_NAME_MAX_SIZE`
```

Is this a general issue or can the macro be fixed to accept constants as args?

Reply:
Easy to fix I think. The macro accepts a "literal". But you can change that to something else like "expr" for expression. The complete list of things you can match against in a macro is here: https://doc.rust-lang.org/reference/macros-by-example.html#metavariables

Macros "1.0" or "macros by example" are like a match statement. So `($input:expr, $len:literal)`, will match `str_to_cstr_force(<any expression>, <any literal>)`. You would want `($input:expr, $len:expr)` I think.

Reply:
The expression must be constant at compile time though, because that number is later used to set the size of an array: `let mut buf = [0u8; $len + 1];`.

Reply:
great, that worked perfectly. thanks!